File: backend/src/main/java/br/com/sgc/service/CompetencyService.java
package br.com.sgc.service;

import br.com.sgc.service.dto.CompetencyDto;
import br.com.sgc.service.dto.DropdownCategoryDto;
import br.com.sgc.service.dto.filter.CompetencyFilterDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface CompetencyService {

    Page<CompetencyDto> findAll(Pageable pageable);

    Page<CompetencyDto> globalSearchFilter(Pageable pageable, String query);

    List<DropdownCategoryDto> findAllDropdown();

    CompetencyDto findById(Long idCompetency);

    CompetencyDto create(CompetencyDto competencyDto);

    CompetencyDto update(CompetencyDto competencyDto);

    void deleteById(Long idCompetency);

    Page<CompetencyDto> columnsFilter(Pageable pageable, CompetencyFilterDto filter);

}

File: frontend/src/app/modules/competency/services/competency.service.ts
import { DropdownModel } from './../../../shared/models/dropdown.model';
import { CompetencyModel } from './../models/competency.model';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { CompetencyFilterModel } from '../models/competency-filter.model';

@Injectable({
  providedIn: 'root'
})
export class CompetencyService {

  constructor(private httpClient: HttpClient) { }

  url: string = 'http://localhost:8080/api/competencia';

  findAll(size: number, page: number): Observable<CompetencyModel[]> {
    const params = { page: page, size: size };
    return this.httpClient.get<CompetencyModel[]>(this.url, { params });
  }

  findAllDropDown(): Observable<DropdownModel[]> {
    return this.httpClient.get<DropdownModel[]>(this.url + `/competencias-dropdown`);
  }

  findById(idCompetency: number): Observable<CompetencyModel> {
    return this.httpClient.get<CompetencyModel>(this.url + '/' + idCompetency);
  }

  create(competencyModel: CompetencyModel): Observable<CompetencyModel> {
    return this.httpClient.post<CompetencyModel>(this.url, competencyModel);
  }

  update(competencyModel: CompetencyModel): Observable<CompetencyModel> {
    return this.httpClient.put<CompetencyModel>(this.url, competencyModel);
  }

  deleteById(idCompetency: number): Observable<void> {
    return this.httpClient.delete<void>(this.url + '/' + idCompetency);
  }

  columnsFilter(filter: CompetencyFilterModel, size: number, page: number): Observable<CompetencyModel> {
    const params = { page: page, size: size };
    return this.httpClient.post<CompetencyModel>(this.url + '/filter', filter, { params });
  }

}
